apps/web/app/api/v1/client/[environmentId]/in-app/sync/route.ts

```diff
@@ -56,6 +59,28 @@ export async function GET(
       const currentResponseCount = await getMonthlyTeamResponseCount(team.id);
       isInAppSurveyLimitReached =
         !hasInAppSurveySubscription && currentResponseCount >= PRICING_APPSURVEYS_FREE_RESPONSES;
+      if (isInAppSurveyLimitReached) {
```

Comment:
As we have duplicated code here between the endpoints I think it makes sense to outsource this to a separate lib file `apps/web/app/api/v1/client/[environmentId]/in-app/sync/lib/posthog.ts`

This also makes the code in the sync endpoints more readable